ui/tests/test_faucet.py

```diff
@@ -59,49 +61,90 @@ def context(
     context.close()
 
 
+def get_metamask_extension_id(context: BrowserContext) -> str:
```

Comment:
```suggestion
from urllib.parse import urlparse

def get_metamask_extension_id(context: BrowserContext) -> str:
    for page in context.background_pages:
        url = page.url
        if url.startswith("chrome-extension://"):
            parsed_url = urlparse(url)
            extension_id = parsed_url.netloc
            
             # Verify it's actually MetaMask - check for known MetaMask identifiers
            # This could be checking page content or other MetaMask-specific attributes
            if is_metamask_extension(page):
                return extension_id
                
    raise Exception("MetaMask extension ID not found.")
   

def is_metamask_extension(page) -> bool:
    """Helper function to verify if a page belongs to MetaMask extension"""
    # Implementation depends on how you can identify MetaMask specifically
    return "MetaMask" in page.title() or "MetaMask" in page.content()
 
```

Plus I would add some documentation, at least some basic definition. That would help us to maintain the code.

Reply:
Readme added